data/router.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+
+const Database = require('./db.js');
+
+const router = express.Router();
+
+// handles urls beginning with /posts. 
+// Might change this to something more semantic now that I know what this API does
+// UPDATE: Changed to posts
+
+// /posts
+router.get('/', async (req, res) => {
+    try {
+        const posts = await Database.find(req.query);
+        res.status(200).json(posts);
+    } catch (error) {
+        // log error to database
+        console.log(error);
+        res.status(500).json({ error:"The posts information could not be retrieved." });
+    }
+});
+
+
+// /posts/:id
+router.get('/:id', async (req, res) => {
+    try {
+        const data = await Database.findById(req.params.id);
+        // If the object returned has a length of zero, we know the post doesn't exist. Error instead of returning empty object
+        if (data.length === 0) {  
+            res.status(404).json({ message:"The post with the specified ID does not exist." });
+        } else {
+            res.status(200).json(data);
+        }
+    } 
+    catch (error) {  // log error to database
+        console.log(error);
+        res.status(500).json({ error:"The post information could not be retrieved." });
+    }
+});
+
+
+router.post('/', async (req, res) => {
+    // Couldn't nest the if statement in the try for some reason
+    // Tried alternating between req.body and post inside of the try to no avail
+    // Function kept defaulting to the catch when if inside of the try and I wasn't getting the correct error message
+    if (!req.body.title || !req.body.contents) { 
+        res.status(400).json({ errorMessage:"Please provide title and contents for the post." });
+        return;
+    }
+    try {
+        const post = await Database.insert(req.body);
+        res.status(201).json(post);
+    } 
+    catch (error) { // Log error to db
+        console.log(error);
+        res.status(500).json({ error:"There was an error while saving the post to the database" });
+    }
+});
+
+
+router.delete('/:id', async (req, res) => {
+    try {
+        const count = await Database.remove(req.params.id);
+        // If the post doesn't exist to be deleted, 404. Otherwise, 200 and delete
+        if (count > 0) {
+            res.status(200).end();
```

Comment:
as far as truthiness goes, you could simply check `if (count)` and zero would be falsey, any positive integer (only other possible successful returns) would be truthy

remember that you still want to send something back with the status code, though! although the default behavior due to the provided helper functions is to just return the count of deleted posts, per the readme you want to return the body of the deleted post, which may require an additional Database call somewhere